streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPeek.java

```diff
@@ -24,7 +24,7 @@
 class KStreamPeek<K, V> implements ProcessorSupplier<K, V> {
 
     private final boolean forwardDownStream;
-    private final ForeachAction<K, V> action;
+    protected final ForeachAction<K, V> action;
```

Comment:
Add comment on why we need to make it `protected`, ditto for the `KStreamPeekProcessor` below for making it package-private.